download release history

GitEdvard opened the pull request:
Release history contains the name, date and description of each release
at a given repo. These fields are fetched from the json object from
GitHub.

Intended workflow: After accepting a release, the user goes to GitHub
and fill in the description manually in release-history. After that the
download-release-history command is given.

GitEdvard committed:
download release history

Release history contains the name, date and description of each release
at a given repo. These fields are fetched from the json object from
GitHub.

Intended workflow: After accepting a release, the user goes to GitHub
and fill in the description manually in release-history. After that the
download-release-history command is given.

withrocks commented on release_tools/github.py:
```diff
@@ -88,6 +89,25 @@ def download_archive(self, branch, save_to_path, ball="zipball"):
+              .format(owner=self.owner, repo=self.repo, token=self.access_token_postfix())
+        response = requests.get(url)
+        if response.status_code == 200:
+            print "Writing to file..."
```
It wouldn't hurt if this was changed to the print function (`from __future__ import print_function`) so it's ready for py3. Or change it to `logging.info(...)`?

withrocks commented on release_tools/github.py:
```diff
@@ -88,6 +89,25 @@ def download_archive(self, branch, save_to_path, ball="zipball"):
+                f.write(self._release_history_contents(response.json()))
+            print "done."
+        else:
+            print "Something went wrong, contents cannot be downloaded"
```
Alternatively one could raise an error here and only write this message in the `cli.py`, then the module is easier to use as a library.

withrocks reviewed (approved):
Looks cool. Two comments, let me know if you want to make changes or if I should merge it as it is.

GitEdvard committed:
Use print syntax for python 3.

GitEdvard committed:
Use exception instead of print

Replace a print to an exception in case when download release history
went wrong.

GitEdvard commented:
Ok, I've added to commits for the print syntax, and for changing an error print to an exception.
